File: festival-web/src/main/java/cn/bdqfork/web/RouterAware.java
package cn.bdqfork.web;

import cn.bdqfork.core.exception.BeansException;
import io.vertx.reactivex.ext.web.Router;

/**
 * @author bdq
 * @since 2020/2/2
 */
public interface RouterAware {
    void setRouter(Router router) throws BeansException;
}

File: festival-kotlin-web/src/main/kotlin/cn/bdqfork/kotlin/web/RouterAware.kt
package cn.bdqfork.web

import cn.bdqfork.core.exception.BeansException
import io.vertx.ext.web.Router

/**
 * @author bdq
 * @since 2020/2/2
 */
interface RouterAware {
    @Throws(BeansException::class)
    fun setRouter(router: Router)
}